app/models/manageiq/providers/vmware/infra_manager/refresher.rb

```diff
@@ -108,6 +108,10 @@ def get_vc_data(ems)
           end
         end
 
+        storage_profile_ids = @vc_data[:storage_profile].collect { |uid, _profile| uid }
+        @vc_data[:storage_profile_datastore] = @vi.pbmQueryMatchingHub(storage_profile_ids)
```

Comment:
Haha yeah if it wasn't for [this](https://github.com/ManageIQ/manageiq/pull/9923/files#diff-e2bbcd12142b044308c0c4b006cf96faL53) you could just add these here